azure-pipelines.yml

```diff
@@ -0,0 +1,72 @@
+parameters:
+  - name: sendgridEnvironments
+    type: object
+    default:
+    - nonprod
+    - prod
+
+trigger:
+  - master
+
+pool:
+  vmImage: 'ubuntu-latest'
+
+stages:
+  - ${{ each environment in parameters.sendgridEnvironments }}:
+    - stage: ${{ environment }}
+
+      variables:
+        isMain: $[eq(variables['Build.SourceBranch'], 'refs/heads/master')]
+        location: uksouth    
+        terraformVersion: 0.14.8
```

Comment:
these should be defined outside of the loop they don't change based on the environment
```suggestion
```

Reply:
moved outside loop.